common/src/main/java/com/wynntils/wynn/model/bossbar/TrackedBar.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.bossbar;
+
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.Util;
+import net.minecraft.util.Mth;
+
+public abstract class TrackedBar {
+    public final Pattern pattern;
+    public final BarType type;
+
+    protected float progress = 0;
+    private float targetPercent = 0;
+    private long setTime = 0;
+
+    private boolean rendered = true;
+
+    protected int current = 0;
+    protected int max = 0;
+
+    protected UUID uuid = null;
+
+    TrackedBar(Pattern pattern, BarType type) {
+        this.pattern = pattern;
+        this.type = type;
+    }
+
+    public void onAdd() {}
+
+    public abstract void onUpdateName(Matcher match);
+
+    public boolean isRendered() {
+        return rendered;
+    }
+
+    public void setRendered(boolean rendered) {
+        this.rendered = rendered;
+    }
+
+    public void setProgress(float progress) {
+        this.progress = this.getProgress();
+        this.targetPercent = progress;
+        this.setTime = Util.getMillis();
+    }
+
+    public float getProgress() {
+        long l = Util.getMillis() - this.setTime;
```

Comment:
Are you trying to update progress slowly over 100 ms?

The interplay between progress, targetPercent, setProgress and getProgress is very complex. I've started at it for a while and am still not fully understanding.

Reply:
I am copying minecraft's `LerpingBossEvent` It lerps from `progress` to `targetProgress`